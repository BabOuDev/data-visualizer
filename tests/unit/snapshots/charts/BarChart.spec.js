import {shallowMount} from '@vue/test-utils';
import BarChart from '@/components/charts/BarChart.vue';

describe('BarChart.vue', () => {
  it('should renders correctly', () => {
    const component = shallowMount(BarChart, {
      props: {
        rows: [
          {colName: 'A'},
          {colName: 'B'},
          {colName: 'B'},
          {colName: 'C'},
          {colName: 'C'},
          {colName: 'C'},
        ],
        col: {path: 'colName'},
      },
    });
    expect(component.html()).toMatch('A');
    expect(component.html()).toMatch('B');
    expect(component.html()).toMatch('C');
    expect(component.html()).toMatchSnapshot();
  });
});
