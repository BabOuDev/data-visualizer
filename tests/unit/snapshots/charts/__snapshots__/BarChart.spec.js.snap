// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarChart.vue should renders correctly 1`] = `
<div class="bar-container">
  <legend-stub keys="A,B,C" colors="#42b883,#3c8171,#35495e"></legend-stub>
  <div class="bar-background"></div>
  <div class="bar-origin"></div>
  <div class="content"><b></b></div>
  <div style="width: 45px; left: 0px; height: 40px; bottom: 0px; background-color: rgb(66, 184, 131);" class="bar" title="A: 1"></div>
  <div style="width: 45px; left: 50px; height: 80px; bottom: 0px; background-color: rgb(60, 129, 113);" class="bar" title="B: 2"></div>
  <div style="width: 45px; left: 100px; height: 120px; bottom: 0px; background-color: rgb(53, 73, 94);" class="bar" title="C: 3"></div>
</div>
`;
